### ISSUE TITLE
Android asset overriding doesn't seem to work

### ISSUE THREAD

**Issue by lizerome** [renpy/renpy#1776]
7.2.0 introduced the ability to override assets on Android by placing files into the game's public directory on the SD card. This should presumably be `/mnt/sdcard/Android/data/org.package.name/files/` with the standard Ren'Py "game", "saves", etc. directories inside that, although [the changelog](https://www.renpy.org/doc/html/changelog.html#android-changes) lists the path as `/mnt/sdcard/Android/package/files/game` instead. I downloaded the stable release of 7.2.1 and packaged "The Question" without touching anything in order to test this feature, and placed an edited version of `bg lecturehall.jpg` under both
> /mnt/sdcard/Android/data/org.renpy.thequestion/files/game/images/bg lecturehall.jpg

> /mnt/sdcard/Android/org.renpy.thequestion/files/game/images/bg lecturehall.jpg

since I wasn't sure which one would work. To my surprise however, **neither** of them worked. After making sure I got the paths right and trying to get things to work for an hour, I decided to go through the recent changes to see where and how the feature was actually implemented.

https://github.com/renpy/renpy/blob/7b2f7d7e8562a59c852c65a38a1887518acf4b33/renpy/main.py#L260-L267

It seems like Ren'Py just checks if the ANDROID_PUBLIC environment variable exists, appends /game to it and then adds that to the search path. ~~Which then lead me to checking what the contents of that variable are, at which point I found that ANDROID_PUBLIC points to... nothing, because it doesn't exist.~~

~~I thought "fine" and decided to simply hardcode the path for the game's public directory, which once again did absolutely nothing. I know for a fact `renpy.android` DOES get evaluated to True on Android and False on my Windows build, and trying to force that path to be added on Windows returns an error before the game can even launch, so this should work. That path also gets automatically created on the application's first run and Ren'Py successfully creates and populates a `saves` directory inside it, as well as a copy of `log.txt` and `traceback.txt` (so despite everything, some "write to the public directory" call somewhere is working just fine).~~

~~Stranger still, after looking into things further, I tried manually printing the output of os.environ[0] and renpy.config.searchpath[0] from inside the game, both of which worked as expected on Windows but return an out of range error on Android, as if their contents are empty as well.~~

Debugging this thing without even having access to the internal Ren'Py console is an absolute nightmare (I tried pressing Shift+O/Shift+D via Hacker's Keyboard and an external one, it ignores everything) and I have no idea where to even go from here. My phone is running stock LineageOS without anything crazy, though I don't have any other device at hand to test this with at the moment.

Any suggestions or advice would be welcome.

***
**Update 1:**
It looks like Ren'Py strips out all variables under the renpy.____ scope on Android while the game is running (presumably for optimization/memory reasons), which explains why I couldn't print any of them. Writing them out to the log right after they're updated works fine however, which produces the following:

![4](https://user-images.githubusercontent.com/48659061/54549050-88818100-49a9-11e9-892d-f501bbf420b3.png)

```
Mon Mar 18 18:16:24 2019
Unknown platform.
Ren'Py 7.2.1.457
 

Bootstrap to the start of init.init took 0.00s
Bootstrap to the start of init.init took 0.00s
Manufacturer Xiaomi model Mi-4c
Screen diagonal is 5.00660720473 inches.
Early init took 1.62s
Early init took 1.62s
-- android_game is: /storage/emulated/0/Android/data/org.renpy.thequestion/files/game
-- added android_game to the search path
------------SEARCH PATH---------------
/storage/emulated/0/Android/data/org.renpy.thequestion/files/game
/data/data/org.renpy.thequestion/files
------------ENVIRONMENT---------------
ANDROID_APP_PATH                 /data/user/0/org.renpy.thequestion/files
EXTERNAL_STORAGE                 /sdcard
SYSTEMSERVERCLASSPATH            /system/framework/org.cyanogenmod.platform.jar:/system/framework/org.cyanogenmod.hardware.jar:/system/framework/services.jar:/system/framework/ethernet-service.jar:/system/framework/wifi-service.jar
BOOTCLASSPATH                    /system/framework/core-oj.jar:/system/framework/core-libart.jar:/system/framework/conscrypt.jar:/system/framework/okhttp.jar:/system/framework/core-junit.jar:/system/framework/bouncycastle.jar:/system/framework/ext.jar:/system/framework/framework.jar:/system/framework/telephony-common.jar:/system/framework/voip-common.jar:/system/framework/ims-common.jar:/system/framework/apache-xml.jar:/system/framework/org.apache.http.legacy.boot.jar:/system/framework/telephony-ext.jar
PATH                             /sbin:/vendor/bin:/system/sbin:/system/bin:/system/xbin
ANDROID_STORAGE                  /storage
ANDROID_BOOTLOGO                 1
ANDROID_ASSETS                   /system/app
ANDROID_PUBLIC                   /storage/emulated/0/Android/data/org.renpy.thequestion/files
ANDROID_SOCKET_zygote_secondary  8
TERMINFO                         /system/etc/terminfo
PYTHONOPTIMIZE                   2
ANDROID_PRIVATE                  /data/user/0/org.renpy.thequestion/files
PYTHONPATH                       /data/user/0/org.renpy.thequestion/files:/data/user/0/org.renpy.thequestion/files/lib
ANDROID_DATA                     /data
ANDROID_APK                      /data/app/org.renpy.thequestion-1/base.apk
ANDROID_OLD_PUBLIC               /storage/emulated/0/org.renpy.thequestion
LD_PRELOAD                       libsigchain.so
DOWNLOAD_CACHE                   /cache
RENPY_RENDERER                   gl
ANDROID_ROOT                     /system
PYTHONHOME                       /data/user/0/org.renpy.thequestion/files
TEST_ENV_VAR                     The test worked.
ASEC_MOUNTPOINT                  /mnt/asec
ANDROID_CACHE                    /cache
ANDROID_ARGUMENT                 /data/user/0/org.renpy.thequestion/files
--------------------------------------
Loader init took 0.01s
Loader init took 0.01s
Loading error handling took 0.27s
Loading error handling took 0.27s
Loading script took 3.04s
Loading script took 3.04s
Saving to /storage/emulated/0/Android/data/org.renpy.thequestion/files/saves
Loading save slot metadata. took 0.01s
Loading save slot metadata. took 0.01s
```

log_clock() logs everything twice on Android for some reason, that's not my doing. Also, ANDROID_PUBLIC only gets set after the initial startup process so it doesn't show up under `/proc/1234/environ`, but it DOES get picked up and handled by Ren'Py perfectly fine. Apologies if this is all painfully obvious, I don't know jack about Ren'Py or Android.

**tl;dr - false alarm, everything seems to be in order with the search path. On the other hand, I'm no closer to figuring out why assets don't get picked up from the public directory.**

***
**Update 2:** No idea whether this is intended or not, but adding files to the other directory in the search path (`/data/data/<package>/files/`) does nothing either. Tried it both with and without a game parent directory, overriding vs. only having the image in the replacement location and filenames with or without spaces.

![5](https://user-images.githubusercontent.com/48659061/54562895-356af680-49c8-11e9-9ee5-5faaffbe2c75.png)

I also tried using an arbitrary path, which works as expected on Windows and does nothing on Android.

![6](https://user-images.githubusercontent.com/48659061/54564079-30f40d00-49cb-11e9-98f8-5b076fc69b8e.png)

**Comment by lizerome** [comment]
Update: Everything is exactly the same on an emulator

![3](https://user-images.githubusercontent.com/48659061/54501126-9c829f80-4923-11e9-8ce2-de123e2c0626.png)

**Comment by renpytom** [comment]
This was fixed by 9608615542668cd5858cd93d7b5fb06cdd303ea0 and 9ddff5d7a5c39298b20cb5f5cdad705b310dd509 .
